README.md

```diff
@@ -582,6 +584,47 @@ Otherwise, instead of raising an error, null values will be coerced as follows:
 - `string` -> `""`
 - `boolean` -> `false`
 
+<a name="largearrays"></a>
+#### Large Arrays
+
+Large arrays are, for the scope of this document, defined as arrays containing
+`20000` items or more.
+
+At some point the overhead caused by the default mechanism used by
+`fast-json-stringify` to handle arrays starts increasing exponentially, leading
+to slow overall executions.
+
+In order to improve that the user can set the `largeArrayMechanism` option with
+one of the following values:
+
+- `default` - Default behavior
+- `json-stringify` - This option will remove support for schema validation
+within **large arrays** completely. By doing so the overhead previously
+mentioned is nulled, greatly improving execution time. Mind there's no change
+in behavior for arrays with less than `20000` items
+- `array-join` - This option is a compromise between the last two.
+`fastify-json-stringify` works by concatenating lots of string pieces into the
+final JSON string. With this option set, **large arrays** would be stringified
+by joining their elements' stringified versions using `Array.join`, instead
+of string concatenation
+
+##### Benchmarks
+
+For reference, here goes some benchmarks for comparison over the three
+mechanisms. Benchmarks conducted on an old machine.
+
+- Machine: `ST1000LM024 HN-M 1TB HDD, Intel Core i7-3610QM @ 2.3GHz, 12GB RAM, 4C/8T`.
+- Node.js `v16.13.1`
+
+```
+JSON.stringify large array x 157 ops/sec ±0.73% (86 runs sampled)
+fast-json-stringify large array default x 48.72 ops/sec ±4.92% (48 runs sampled)
+fast-json-stringify large array json-stringify x 157 ops/sec ±0.76% (86 runs sampled)
+fast-json-stringify large array array-join x 69.04 ops/sec ±4.47% (53 runs sampled)
+compile-json-stringify large array x 175 ops/sec ±4.47% (79 runs sampled)
```

Comment:
why compile-json-stringify does not suffer the penalty? Maybe there is something we can learn?

Reply:
`compile-json-stringify` differs from `fast-json-stringify` in the fact that this lib uses the schema to validate the data output whereas that one validates input.

On stringifying, CJS relies on `JSON.stringify` for individual elements and on string concatenation for gluing things together. There's not much overhead in that process. FJS' overhead comes from the data validation that's done on the stringify process.

If we want to compare with them we need to do so with the `json-stringify` proposed mechanism. Even then I believe it's expected that FJS performs a bit worse because the data validation is only missing from large arrays and its elements. It's still there for all other data.